eru/src/utils/index.js

```diff
@@ -41,3 +41,9 @@ export const isWithinTenMinutes = (dbTimestamp) => {
   const tenMinutesInMilliseconds = 10 * 60 * 1000; //the 10 is ten minutes
   return difference <= tenMinutesInMilliseconds;
 };
+
+let debounceTimer;
+export const debounce = (callback, time) => {
```

Comment:
what is the role of debounce? please can you add more comments?

Reply:
Normally each time a user types a character in the tag input field a request is made to the backend to search the tags which match it. So i used debounce to limit how often that function is called so that a fewer number of requests will be made to the backend.

Reply:
let me add comments in that section

Reply:
> Normally each time a user types a character in the tag input field a request is made to the backend to search the tags which match it. So i used debounce to limit how often that function is called so that a fewer number of requests will be made to the backend.

so it'll not limited the user accessibility?

Reply:
no it will not. It just reduces the number of request sent to the backend. The final search request is sent only when the user stops typing for 1.5 seconds. It is a technique used in most popular websites to reduce traffic on the server during search.

Reply:
you can do some research about the debounce function to know more about it. It is very popular

Reply:
so in your opinion, if I try tag like "vuejs", the endpoint will be hit how many times?

Reply:
> no it will not. It just reduces the number of request sent to the backend. The final search request is sent only when the user stops typing for 1.5 seconds. It is a technique used in most popular websites to reduce traffic on the server during search.
1.5 seconds is not too much?

Reply:
> so in your opinion, if I try tag like "vuejs", the endpoint will be hit how many times?

if you type vuejs in less than 1,5s the endpoint will be hit one time instead of 5 times

Reply:
> > no it will not. It just reduces the number of request sent to the backend. The final search request is sent only when the user stops typing for 1.5 seconds. It is a technique used in most popular websites to reduce traffic on the server during search.
> 
> 1.5 seconds is not too much?

Well we can adjust the timeout to optimize user experience

Reply:
> > so in your opinion, if I try tag like "vuejs", the endpoint will be hit how many times?
> 
> if you type vuejs in less than 1,5s the endpoint will be hit one time instead of 5 times

great, good job

Reply:
> 

good job about research on "debounce"

Reply:
Thanks! you can test it and see. Then we can adjust the timeout for the debounce